policy-helper.c

```diff
@@ -28,7 +28,7 @@ void helper_validator_validate(CPURISCVState *env, target_ulong pc, uint32_t opc
    skipped_commit = true;
 
    policy_validator_hack_env = env;
-   if (!e_v_validate(pc, opcode)) {
+   if (!e_v_validate((uint64_t)pc, opcode)) {
```

Comment:
Won't this cast differ by architecture?

Reply:
Yes, but that should all be taken care of by the compiler/linker.  For RV32 it will be put into two registers and linked against the 32-bit policy engine, and for RV64 it will be put into a single register and linked against the 64-bit policy engine.